### PR TITLE
Studio clock

### LINKED ISSUE #662: Studio View - On Air does not change to reflect OnTime status
The 'On Air' in the Studio view does not reflect the status of OnTime.

I think this is due to both states being set to the same colour:

color: var(--studio-idle, #c53030);
color: var(--studio-active, #c53030);

Version 2.24.8 in Docker on Raspberry Pi 
Studio View checked on Safari (17.2) and Firefox (120.0.1) on MacOS (14.2)

### PR DESCRIPTION
**Pull request by alex-Arc**
Fixes #662  and #97

### CODE REVIEW

**Comment by cpvalente** [comment]
Very impressive, thank you
I wonder if the test fields on the right side should all be in "normal" font? 

That would increase readability and solve the issues with character support

What do you think?

File: apps/client/src/features/viewers/studio/StudioClock.scss [hunk 1]
```diff
@@ -123,23 +157,23 @@ $cyan-idle: #0aa;
...
       padding-bottom: 2vh;
       font-size: 15vh;
-      line-height: 0.9em;
+      line-height: 1.2em;
       color: var(--studio-active, $red-active);
 
       &--idle {
-        color: var(--studio-idle, $red-active);
+        color: $backlight;
+        // color: var(--studio-idle, $red-idle);
```
Comment (cpvalente) [inline apps/client/src/features/viewers/studio/StudioClock.scss:170]: Remember to remove unused code
We also need to maintain the variables where possible

Especially if we are merging to v2

File: apps/client/src/features/viewers/studio/StudioClock.tsx [hunk 2]
```diff
@@ -83,16 +85,28 @@ export default function StudioClock(props: StudioClockProps) {
       <NavigationMenu />
       <ViewParamsEditor paramFields={studioClockOptions} />
       <div className='clock-container'>
-        <div className={`studio-timer ${showSeconds ? 'studio-timer--with-seconds' : ''}`}>{clock}</div>
+        <div className='studio-timer-warp'>
```
Comment (cpvalente) [inline apps/client/src/features/viewers/studio/StudioClock.tsx:88]: Question: what does timer-warp mean?

File: apps/client/src/features/viewers/studio/StudioClock.tsx [hunk 3]
```diff
@@ -83,16 +85,28 @@ export default function StudioClock(props: StudioClockProps) {
...
+        <div className='next-countdown-warp'>
+          <div
+            className={cx([
+              'next-countdown',
+              isNegative ? ' next-countdown--overtime' : '',
+              isPaused ? ' next-countdown--paused' : '',
+            ])}
+          >
+            {/* ! is used an a blank character //TODO: placeholder moves a bit when becomming active*/}
+            {isNegative ? '-' : '!'}
```
Comment (cpvalente) [inline apps/client/src/features/viewers/studio/StudioClock.tsx:107]: I wonder if it is best not to have the `-` symbol. This would mean the test moves a bit, but that is maybe OK.
Most of the time the `-` will not be there, and the clock will be off-centre, so I wonder if we are sacrificing the common case for an edge case

File: apps/client/src/features/viewers/studio/StudioClock.scss [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+  src:
+    local('DSEG14'),
+    url('./../../../assets/fonts/DSEG14ClassicMini-Regular.ttf') format('truetype');
+}
+
+@font-face {
+  font-family: 'mono';
+  src:
+    local('MONO'),
+    url('./../../../assets/fonts/Hack-Regular.ttf') format('truetype');
```
Comment (cpvalente) [inline apps/client/src/features/viewers/studio/StudioClock.scss:21]: question: could we use a default monospace font to avoid one more import? I dont know in terms of style if this one is significantly better

Comment (alex-Arc) [reply apps/client/src/features/viewers/studio/StudioClock.scss:21]: yeah.. i dont like the default monospace that much so i'm trying to finde one that fits better with the lowtech look

File: apps/client/src/features/viewers/studio/StudioClock.tsx (hunk 2 above)
Comment (alex-Arc) [reply apps/client/src/features/viewers/studio/StudioClock.tsx:88]: the warp is used to create the "backlight" effect

**Review by cpvalente — COMMENTED** [review]
made a few suggestions. what do you think?

**Comment by cpvalente** [comment]
@alex-Arc , what do you think about the state of this?

File: apps/client/src/features/viewers/studio/StudioClock.scss [hunk 5]
```diff
@@ -1,182 +1,198 @@
...
-    margin: 4vh 0;
-    font-family: digital-clock, monospace;
-    text-transform: uppercase;
+.next-title {
+  font-family: monospace;
+  font-weight: 400;
+  color: var(--studio-active-label, $cyan-active);
+  font-size: calc(var(--clock-size) / 10);
+  text-align: center;
+  overflow: clip visible;
```
Comment (alex-Arc) [inline apps/client/src/features/viewers/studio/StudioClock.scss:116]: should we change this to 
`overflow: clip hidden;` just to catch the edge case

image

File: apps/client/src/features/viewers/studio/StudioClock.scss [hunk 6]
```diff
@@ -1,182 +1,198 @@
...
 
-      &--idle {
-        color: var(--studio-idle, $red-active);
-      }
-    }
+  &--overtime {
+    color: var(--studio-active, $red-active);
+  }
+  &--paused {
+    color: var(--studio-active, $orange-active);
```
Comment (alex-Arc) [inline apps/client/src/features/viewers/studio/StudioClock.scss:128]: We should add the pause orange to the style override

File: apps/client/src/features/viewers/studio/StudioClock.scss [hunk 7]
```diff
@@ -1,182 +1,198 @@
...
 /* =============== CLOCK STUFF ==================*/
 
-$clock-size: 90vh;
-$size-hours: min(3vh, 20px);
-$half-hours: min(1.5vh, 10px);
-$size-min: min(2.5vh, 18px);
-$half-min: min(1.25vh, 9px);
-$red-active: #c53030;
-$red-idle: #000000;
+$size-hours: clamp(12px, 2.5vw, 22px);
```
Comment (alex-Arc) [inline apps/client/src/features/viewers/studio/StudioClock.scss:13]: don't know if we want to fix this but we only scale down the dots proportional to the width and not the height

image

**Comment by alex-Arc** [comment]
am/pm indicator?
image

**Review by cpvalente — COMMENTED** [review]
good tips, thank you @alex-Arc , let me know if you think this is good enough for now.
We can revisit later

**Comment by alex-Arc** [comment]
Yes this is good
